src/test/java/org/opensearch/security/IndexTemplateClusterPermissionsCheckTest.java

```diff
@@ -36,10 +36,10 @@ public void setupRestHelper() throws Exception{
         }
         @Test
         public void testPutIndexTemplateByNonPrivilegedUser() throws Exception {
-            String expectedFailureResponse = getFailureResponseReason("ds3");
+            String expectedFailureResponse = getFailureResponseReason("ds4");
 
             // should fail, as user `ds3` doesn't have correct permissions
-            HttpResponse response = rh.executePutRequest("/_index_template/sem1234", indexTemplateBody, encodeBasicHeader("ds3", "nagilum"));
+            HttpResponse response = rh.executePutRequest("/_index_template/sem1234", indexTemplateBody, encodeBasicHeader("ds4", "nagilum"));
```

Comment:
`ds4` has only `"DATASTREAM_ALL"` permission, but `ds3` has more permissions. Why does this change allow `ds3` to put this index template but they wasn't allowed to do?

Reply:
This commit changed permissions for `ds3` https://github.com/opensearch-project/security/commit/f7cc569c9d3fa5d5432c76c854eed280d45ce6f4#diff-2ffb4c04831d49dff0d81bdee2221ecdda9fd1d02632c0a4322419b0d2f68319L1125

I created a new user `ds4` with the original expected permissions for `ds3`